### DISCUSSION TITLE
Gradient in `MiniMap`?

### DISCUSSION THREAD

**Discussion by CSworenI** [xyflow/xyflow#4582] [Q&A] [answered]
I have a custom decision node which is diamond shaped (which I figured out can be persisted using the `nodeClassName` attribute), but when a decision hasn't been reached, I apply a class of `undecided` which looks like this:
```
.undecided { background: linear-gradient(90deg, #34d399 50%, white 20%, #ffa3a6 40%); }
```

I've tried applying it directly in the `MiniMap.nodeColor` attribute, but it comes out a light grey (I assume the default node color), rather than the watermelon-y green-to-pink gradient that I want to see. Is there a way to apply this gradient?

Here is my current `MiniMap` implementation:
```
{miniMap && <MiniMap
	nodeColor={(n) => n.type === 'process' ? n.style.background : COLORS[n.data.class] ?? 'linear-gradient(90deg, #34d399 50%, white 20%, #ffa3a6 40%)' }
	nodeBorderRadius={n => ['START', 'END'].indexOf(n.data.status) > -1 ? 0 : 5}
	nodeClassName={n => n.type === 'decision' && 'diamond' }
	onNodeClick={onNodeClick}
	position={miniMapPos}
	pannable={true}
/>}
```

As has been suggested for inline styles in React, I have removed the semicolon from the gradient, which applies it to the node everywhere else. Perhaps I'm just missing a critical part of the documentation which might cover this kind of thing. Let me know if it can't currently be done.

**Comment by bcakmakoglu** [comment]
When you inspect the minimap nodes you'll see that the color is assigned to it using `fill: color`.
`fill` does not accept a linear-gradient the way `background` does and `background` will have no effect on an svg rect, which minimap nodes are.

You'll have to define an svg gradient like this
```html
<svg style={{ height: 0 }}>
  <defs>
    <linearGradient id="watermelon" x1="0%" y1="0%" x2="100%" y2="0%">
      <stop offset="50%" stop-color="#34d399" />
      <stop offset="70%" stop-color="white" />
      <stop offset="110%" stop-color="#ffa3a6" />
    </linearGradient>
  </defs>
</svg>
```

and then use `fill: url(#watermelon)`

Here's an example: https://codesandbox.io/p/sandbox/winter-sun-g4vy7s?file=%2Fsrc%2FApp.tsx%3A28%2C7-36%2C13

**Reply by CSworenI** [reply]
@bcakmakoglu When I [alter your sandbox to include my styles](https://codesandbox.io/p/sandbox/elated-resonance-v8kxmk), your 'position-logger' disappears completely. When I do so in my local, the minimap still does not give the gradient (or any color at all) to the 'undecided' decision node(s).
![image](https://github.com/user-attachments/assets/0373811b-5e93-4f24-a60e-b35f868c438c)

**Reply by bcakmakoglu** [reply, marked as answer]
There's a number of issues with the CSS you copy pasted into there.

[Here's a corrected sandbox.](https://codesandbox.io/p/sandbox/wonderful-breeze-ctkxrv?file=%2Fsrc%2Fstyles.css%3A16%2C16)

**Reply by CSworenI** [reply]
@bcakmakoglu Thanks, that gets me there for the `MiniMap`. Let's continue with the [resizer issue](https://github.com/xyflow/xyflow/discussions/4534) I'm running into...
